pkg/status/controller.go

```diff
@@ -140,6 +286,18 @@ func (c *Controller) runWorkStatusInformer(ctx context.Context) {
 		Resource: util.WorkStatusResource}
 
 	c.workStatusInformer = informerFactory.ForResource(gvr).Informer()
+	// add indexer on key from (wecName, sourceRef) for workstatus fetching efficiency
+	c.workStatusInformer.AddIndexers(cache.Indexers{
+		workStatusIdentificationIndexKey: func(obj interface{}) ([]string, error) {
+			wecName := obj.(metav1.Object).GetNamespace()
+			sourceRef, err := util.GetWorkStatusSourceRef(obj.(runtime.Object))
+			if err != nil {
+				return nil, err
```

Comment:
Fun fact: the informer library code will panic if this func returns a non-nil error.